cumulus_library/template_sql/utils.py

```diff
@@ -0,0 +1,131 @@
+"""utility functions related to populating jinja templates
+
+Currently, these deal with various edge cases around complex nested objects in sql
+and validating them so that the actual generated queries can be constructed more
+simply. This includes, but is not limited, to:
+    - Data missing entirely
+    - Data present, but 'nullish' - with some structure but no value
+    - Data with deep missing elements
+    - Data which may or may not be in an array depending on context
+"""
+from rich.progress import Progress, Task
+
+from cumulus_library.template_sql.templates import (
+    get_column_datatype_query,
+    get_is_table_not_empty_query,
+)
+
+
+def is_codeable_concept_populated(
+    schema: str,
+    table: str,
+    base_col: str,
+    cursor,
+    progress: Progress,
+    task: Task,
+    coding_element="coding",
+) -> bool:
+    """Check db to see if codeableconcept data exists.
+
+    Will execute several exploratory queries to see if the column in question
+    can be queried naively. Will advance the associated progress's task by 3 steps.
+
+    :param schema: The schema/database name
+    :param table: The table to query against
+    :param base_col: the place to start validation from.
+        This can be a nested element, like column.object.code
+    :param cursor: a PEP-249 compliant database cursor
+    :param progress: a Rich progress bar manager
+    :param task: The TaskID of the progress bar to update
+    :param coding_element: the place inside the code element to look for coding info.
+        default: 'coding' (and :hopefully: this is always right)
+    :returns: a boolean indicating if valid data is present.
+    """
+    query = get_is_table_not_empty_query(table, base_col)
+    cursor.execute(query)
+    progress.advance(task)
+    if cursor.fetchone() is not None:
```

Comment:
nit: you could avoid indenting the whole function if you just early-exited instead and flipped this condition to `if not cursor.fetchone(): advance; return;` (and same comment later for the second logic branch)

That (to me) would make this more readable instead of a tree of "OK which path am I in?". This feels like a brain difference though, because I feel like we've talked about early-exiting vs if-trees before.

Reply:
i think in this case it's because of the advance=2 call, which, per the other comment, if we treat this as atomic early exiting is fine.

Reply:
Well regardless of atomic or not, you can just do:
```
if not cursor.fetchone():
    advance
    return
```

Reply:
oh of course

i'm still pulling it out, i think tracking the iterations across files is too much.